Validation/EcalDigis/src/EcalDigisValidation.cc

```diff
@@ -306,13 +305,12 @@ void EcalDigisValidation::analyze(edm::Event const & e, edm::EventSetup const &
   if ( isEndcap ) {
 
     e.getByToken( crossingFramePCaloHitEEToken_, crossingFrame );
-    std::auto_ptr<MixCollection<PCaloHit> > 
-      endcapHits( new MixCollection<PCaloHit>( crossingFrame.product() ) );
+    const MixCollection<PCaloHit> &endcapHits = crossingFrame.product();
```

Comment:
Maybe it would be clearer for future maintenance to explicitly show you created a new `MixCollection`?
```C++
const MixCollection<PCaloHit> endcapHits{crossingFrame.product()};
```